tests/spec/int_series_ops/test_division_ops.py

```diff
@@ -0,0 +1,66 @@
+from ..tables import p
+
+title = "Arithmetic division operations"
+
+table_data = {
+    p: """
+          |  i1 |  i2
+        --+-----+------
+        1 | 101 | 111
+        2 |  -1 |   2
+        3 |  -4 |   3
+        4 |     | 201
+        """,
+}
+
+
+def test_truedivide(spec_test):
+    spec_test(
+        table_data,
+        p.i1 / p.i2,
+        {
+            1: 101 / 111,
+            2: -1 / 2,
+            3: -4 / 3,
+            4: None,
+        },
+    )
+
+
+def test_truedivide_by_constant(spec_test):
+    spec_test(
+        table_data,
+        p.i1 / 10,
+        {
+            1: 101 / 10,
+            2: -1 / 10,
+            3: -4 / 10,
+            4: None,
+        },
+    )
+
+
+def test_floordivide(spec_test):
+    spec_test(
+        table_data,
+        p.i1 // p.i2,
+        {
+            1: 101 // 111,
+            2: 0,
+            3: -1,
```

Comment:
These two lines test the following:
`-1  // 2 == 0`
`-4 // 3 == -1`

Usually we specify the arithmetic operation here in the spec tests, but if we expect different behaviours between  in_memory and SQL (sqlite, mssql) I'm not sure what the best approach is.

Depending on whether I specify the results that I expect from in_memory (`-1  // 2 == -1` and `-4 // 3 == -2`) or SQL (see above) I get passed test for one or the other.

I thought that defining a `floordiv` function that matches the SQL behaviour (see [here](https://github.com/opensafely-core/databuilder/pull/920/files#r1095765003)) would fix all my problems but so far I couldnt get that to work.

Reply:
We definitely don't expect different behaviour between the in-memory and SQL engines. That's part of the point of the spec tests! As for defining a Python function which matches the SQL behaviour, they behave the same when the result is positive and only diverge for negative results. So how about:
```py
def floordiv(lhs, rhs):
    result_is_negative = (lhs < 0) != (rhs < 0)
    absolute_result = abs(lhs) // abs(rhs)
    return -absolute_result if result_is_negative else absolute_result
```

Reply:
thanks! I think this works everywhere but some tests still failing (but I _think_ because of other problems, mainly my not ideal choices for test data and how I specify the actual and expected results)

Reply:
following the tests from date series ops I noticed that I need to specify the actual result that I'm expecting when the engines differ (e.g., [date_series_ops/test_date_series_ops.py](https://github.com/opensafely-core/databuilder/blob/112e6cd91a4cdc1c40a0c080d964e990dcd96028/tests/spec/date_series_ops/test_date_series_ops.py#L197-L209)), instead of rewriting the operation (e.g., [float_series_ops/test_arithmetic_ops.py](https://github.com/opensafely-core/databuilder/blob/112e6cd91a4cdc1c40a0c080d964e990dcd96028/tests/spec/float_series_ops/test_arithmetic_ops.py#L32-L36))